### PR TITLE
Adjustment practice

### PR DESCRIPTION
**Pull request by Jiayue-He**

### CODE REVIEW

File: app.R [hunk 1]
```diff
@@ -0,0 +1,1215 @@
...
+#library(car)
+#library(rgdal)
+library(shinyWidgets)
+library(ggplot2)
+library(plotly)
+library(rlocker)
+
+library(dplyr)
+library(shinycssloaders)
+
```
Comment (shravanisamala) [inline app.R:24]: Missing the BoastUtils package! When I tried running this code on my laptop, it says that the script function which is in BoastUtils was not found.

File: app.R [hunk 2]
```diff
@@ -0,0 +1,1215 @@
...
+source("helpers.R") 
+
+ui <- dashboardPage(
```
Comment (shravanisamala) [inline app.R:49]: According to the style guide, I believe this should be:

ui <- list(
     dashboardPage( ...

File: app.R [hunk 3]
```diff
@@ -0,0 +1,1215 @@
...
+    sidebarMenu(
+      id = "pages",
+      menuItem("Prerequisites", tabName = "prereq", icon = icon("book")),
+      menuItem("Overview",tabName = "instruction", icon = icon("dashboard")),
+      menuItem("Explore",tabName = "explore", icon = icon("wpexplorer")),
+      menuItem("Game", tabName = "qqq", icon = icon("gamepad")),
+      menuItem("References", tabName = "references", icon = icon("leanpub"))
+    ),
+    
+    tags$div(
```
Comment (shravanisamala) [inline app.R:78]: Here is the updated PSU button code from the style guide!

tags$div(
        class = "sidebar-logo",
        boastUtils::sidebarFooter()
      )

File: app.R [hunk 4]
```diff
@@ -0,0 +1,1215 @@
...
+      tags$style(HTML(".js-irs-4 .irs-single, .js-irs-4 .irs-bar-edge, .js-irs-4 .irs-bar {border-color: #ffc04d")),
+      tags$style(HTML(".js-irs-5 .irs-single, .js-irs-5 .irs-bar-edge, .js-irs-5 .irs-bar {border-color: #ffc04d")),
+      tags$style(HTML(".js-irs-6 .irs-single, .js-irs-6 .irs-bar-edge, .js-irs-6 .irs-bar {border-color: #ffc04d")),
+      tags$style(HTML(".js-irs-7 .irs-single, .js-irs-7 .irs-bar-edge, .js-irs-7 .irs-bar {border-color: #ffc04d")),
+      tags$style(HTML(".js-irs-8 .irs-single, .js-irs-8 .irs-bar-edge, .js-irs-8 .irs-bar {border-color: #ffc04d"))
+    ),
+    tabItems(
+      #Adding pre-requisites page to remove background from instructions page
+      
+      tabItem(tabName="prereq",
```
Comment (shravanisamala) [inline app.R]: The tab items are out of order. I think it should be Overview page, then prerequisites page

File: app.R [hunk 5]
```diff
@@ -0,0 +1,1215 @@
...
+      class = "sidebar-logo",
+      boastUtils::sidebarFooter()
+    )
+  ),
+  
+  ####################### Button and slider bar color edits ######################################
+  ######### Could be combined but left separate so easily understood#####################
+  
+  dashboardBody(
+    tags$head(
```
Comment (shravanisamala) [inline app.R]: Not quite sure what lines 61 - 113 are supposed to do so I tried commenting it out and the app's background color for each page remained white! Consider asking Dr. Neil or Mr. Carey if it is ok to completely delete these lines of code.

File: app.R [hunk 6]
```diff
@@ -0,0 +1,1215 @@
...
+      tags$style(HTML(".js-irs-5 .irs-single, .js-irs-5 .irs-bar-edge, .js-irs-5 .irs-bar {border-color: #ffc04d")),
+      tags$style(HTML(".js-irs-6 .irs-single, .js-irs-6 .irs-bar-edge, .js-irs-6 .irs-bar {border-color: #ffc04d")),
+      tags$style(HTML(".js-irs-7 .irs-single, .js-irs-7 .irs-bar-edge, .js-irs-7 .irs-bar {border-color: #ffc04d")),
+      tags$style(HTML(".js-irs-8 .irs-single, .js-irs-8 .irs-bar-edge, .js-irs-8 .irs-bar {border-color: #ffc04d"))
+    ),
+    tabItems(
+      #Adding pre-requisites page to remove background from instructions page
+      
+      tabItem(tabName="prereq",
+              h3(strong("Logistic Regression Analysis")),
```
Comment (shravanisamala) [inline app.R]: Before this, consider adding h2(Prerequisite) because I think it is important that each page have a title.

Comment (shravanisamala) [inline app.R]: Delete strong()

File: app.R [hunk 7]
```diff
@@ -0,0 +1,1215 @@
...
+           cex.main =1.5, pch=16,las=1)  
+    }
+  })
+  
+  ##### goodness of fit#####
+  HLresult<-function(){
+    input$goButton
+    df = isolate(commonDf())
+    mod <- glm (y~x, data=df, family = binomial)
+    hl <- hoslem.test(mod$y, fitted(mod))
```
Comment (shravanisamala) [inline app.R]: For some reason, when I open the run the code everything works but then when I look in the console it says hoslem.test not found. I believe you talked about this with Dr. Neil during the meeting and I google this function and it said there is a package called ResourceSelection that has this function. However, it says that the version of that package is not available for my version of R.

File: app.R [hunk 8]
```diff
@@ -0,0 +1,1215 @@
...
+
+library(dplyr)
+library(shinycssloaders)
+
+source("helpers.R") 
+
+ui <- dashboardPage(
+  skin = 'yellow',
+  dashboardHeader(title = "Logistic Regression",
+                  tags$li(class="dropdown",
```
Comment (shravanisamala) [inline app.R]: Just a small thing, but make sure to add spaces around equal signs. In the beginning, I think we enabled a setting in R that would give us small blue warnings on the side when there were these small style issues present!

File: app.R [hunk 9]
```diff
@@ -0,0 +1,1215 @@
...
+              br(),
+              br(),
+              div(style = "text-align: center",bsButton("start","Go to the overview",
+                                                        icon("bolt"),style = "danger",
+                                                        size = "large",class="circle grow"))
+              
+      ),
+      
+      tabItem(tabName = "instruction",
+
```
Comment (shravanisamala) [inline app.R]: Don't forget to add Logistic Regression as the h1 title. Also, I noticed that the lists are in h4(). I do not think you need this as it makes the font look too big. The style guide usually laid out lists as p( and the tags$li; I would also consider changing this in the prerequisites page as well.

File: app.R [hunk 10]
```diff
@@ -0,0 +1,1215 @@
...
+      ),
+      
+      tabItem(tabName = "instruction",
+              
+              tags$a(href='http://stat.psu.edu/',tags$img(src='logo.png', align = "left", width = 180)),
+              br(),br(),br(),
+              h3(strong("About:")),
+              h4("This app allows you to explore how different factors can affect the outcome of the Logistic Regression Model and Empirical Logit Plot."),
+              br(),
+              h3(strong("Instructions:")),
```
Comment (shravanisamala) [inline app.R]: remove strong from all titles and the color as well

File: app.R [hunk 11]
```diff
@@ -0,0 +1,1215 @@
...
+              h4(tags$li("This app includes Single Logistic Regression with simulated data and the Empirical Logit Plot with real datasets.")),
+              #h4(tags$li("For each model, adjust the sliders to change the sample size and corresponding beta coefficients.")),
+              h4(tags$li("Click New Sample button to generate plot. Watch the change of plot when drag the slider of confidence interval.")),
+              h4(tags$li("In Empirical Logit Plot, select interested predictors from the menu and see how the plot changes accordingly")),
+              h4(tags$li("After working with the Explore section, you can start the game to test your understanding of the concepts.")),
+              h4(tags$li("Practice the questions in Game Section. For each question you get right, you would get a chance to roll the dice.")),
+              h4(tags$li("If the cumulative total for your dice roll reaches 20 within 10 questions, YOU WIN!")),
+              br(),
+              div(style = "text-align: center",
+                  bsButton(inputId = "go", label =  "Explore", icon("bolt"), style= "danger", size= "large", class='circle grow')
```
Comment (shravanisamala) [inline app.R]: Another small thing, but I would indent each thing inside bsButton such as 
inputId = 
label = 
etc. 

Just for code readability purposes

File: app.R [hunk 12]
```diff
@@ -0,0 +1,1215 @@
...
+              br(),
+              h3(strong("About the data:")),
+              h4("The datasets and the procedures for the empirical logit plot are adopted from Stat2: Models for a World of Data 
+                                       by Cannon, Cobb, Hartlaub, Legler, Lock, Moore, Rossman, and Witmer.  ")
+      ),
+      
+      
+      
+      tabItem(tabName = "explore",
+
```
Comment (shravanisamala) [inline app.R]: Add a descriptive title in h2()

File: app.R [hunk 13]
```diff
@@ -0,0 +1,1215 @@
...
+                      ),
+                      
+                      selectInput(
+                        inputId="residualType", 
+                        label = "Residual Type",
+                        choices = c("deviance", "pearson"),
+                        selected="deviance"),
+                      
+                      br(),
+                      actionButton("goButton", "New Sample", icon("paper-plane"),
```
Comment (shravanisamala) [inline app.R]: https://educationshinyappteam.github.io/Style_Guide/htmlCSS.html

I was looking in the style guide (sorry I keep saying style guide I know I am mentioning it a lot!), and I believe there are certain colors you can use for buttons and I would just write out the color if it is one of those option instead of #ffff so future contributors will know what color it is more easily. Also I did this on one of the apps I was reviewing as well, but Dr. Neil said to take out the class = "circle grow".

File: app.R [hunk 14]
```diff
@@ -0,0 +1,1215 @@
...
+  
+  #adding prereq pages and game pages
+  dashboardSidebar(
+    width = 250,
+    sidebarMenu(
+      id = "pages",
+      menuItem("Prerequisites", tabName = "prereq", icon = icon("book")),
+      menuItem("Overview",tabName = "instruction", icon = icon("dashboard")),
+      menuItem("Explore",tabName = "explore", icon = icon("wpexplorer")),
+      menuItem("Game", tabName = "qqq", icon = icon("gamepad")),
```
Comment (shravanisamala) [inline app.R]: tabName could probably be game instead of qqq? I do not know if "qqq" stands for anything important, so if it does do not change it.

File: app.R [hunk 15]
```diff
@@ -0,0 +1,1215 @@
...
+                  bsButton(inputId = "go2", label =  "Continue", icon("bolt"), style= "danger", size= "large", class='circle grow')
+              )
+      ),
+      
+      #References page
+      tabItem(
+        tabName = "references",
+        withMathJax(),
+        h2("References"),
+        p(
```
Comment (shravanisamala) [inline app.R:437]: Make sure you cite all of the packages you used!

File: app.R [hunk 16]
```diff
@@ -0,0 +1,1215 @@
...
+  observeEvent(input$info,{
+    sendSweetAlert(
+      session = session,
+      title = "Instructions:",
+      text = "This app explores Simple Logistic Regression with simulated data and real data",
+      type = NULL
+    )
+  })
+  observeEvent(input$go,{
+    updateTabItems(session,"pages","explore")
```
Comment (shravanisamala) [inline app.R]: change these to 
updateTabItems(
     session = session, 
     inputId = "pages", 
     selected = "explore")

Comment (shravanisamala) [reply app.R]: And for the next five as well!

File: app.R [hunk 17]
```diff
@@ -0,0 +1,1215 @@
...
+  observeEvent(input$goButton, {
+    # When the button is clicked, wrap the code in a call to `withBusyIndicatorServer()`
+    withBusyIndicatorServer("goButton", {
+      Sys.sleep(1)
+    })
+  })
+  
+  ############################Gray out buttons###############################
+  # observeEvent(input$start, {
+  #   updateButton(session, "answer", disabled = TRUE)
```
Comment (shravanisamala) [inline app.R]: same as comment above but, make sure you make each attribute (not sure this is the right word) explicit. Always type out session = session and inputId = "" etc.

File: app.R [hunk 18]
```diff
@@ -0,0 +1,1215 @@
...
+                             analysis.” Journal of Statistical Software, 40, pp. 1-29. 
+                             Available at http://www.jstatsoft.org/v40/i01/."
+        ),
+        
+        p(
+          class = "hangingindent",
+          "Carey, R. (2019). boastUtils: BOAST Utilities, R Package.
+                             Available from https://github.com/EducationShinyAppTeam/boastUtils"
+        )
+      ) #end of tabItem
```
Comment (shravanisamala) [inline app.R]: Don't forget the copyright info at the end of the references page!

        br(),
        br(),
        br(),
        boastUtils::copyrightInfo()

File: app.R [hunk 19]
```diff
@@ -0,0 +1,1215 @@
...
+                Sys.sleep(1)
+                img(src = "26.png",width = '30%')
+              })
+            }
+            
+          }
+          
+          
+        }
+      }
```
Comment (shravanisamala) [inline app.R:956]: Check bracket alignment

File: app.R [hunk 20]
```diff
@@ -0,0 +1,1215 @@
...
+    })
+    
+  })
+  
+  renderIcon()
+  
+
+  observeEvent(input$restart,{
+    updateButton(session, "submit", disabled = FALSE)
+    updateButton(session,"restart",disable =FALSE)
```
Comment (shravanisamala) [inline app.R]: disabled not disable

File: app.R (hunk 16 above)
Comment (shravanisamala) [reply app.R]: Also, when you switch the order of the tabs, these buttons will have to modified! Also, Dr. Pearl once told me that some of my buttons were too wordy when they had, for example, "Go to Prerequisites". He suggested just have the word "Go!" on it but changing the symbol to the page the button would go to. So for example, when you will have the go button on your overview page going to the prerequisite page, it would just say, "GO!" with the book icon next to it!

File: app.R [hunk 21]
```diff
@@ -0,0 +1,1215 @@
...
+    answer<-isolate(input$answer)
+    if (any(answer == ans[value$index,1])){
+      output$dice<-renderUI({
+        img(src = "newdice1.gif", width = '30%')
+      })
+      active(TRUE)
+    }
+    else{
+      if(length(index_list$list) == 1){
+        updateButton(session, "nextq", disabled = TRUE)
```
Comment (shravanisamala) [inline app.R]: session = session etc.

File: app.R [hunk 22]
```diff
@@ -0,0 +1,1215 @@
...
+      }
+      else{
+        HTML(paste("Don't give up, try again !", bank[value$index,7], collapse = "\n")
+        )
+      }
+    })
+    
+  })
+  
+  renderIcon()
```
Comment (shravanisamala) [inline app.R]: I tried commenting this out and the app works without it. Was this something you were working on or can it be deleted?

**Review by shravanisamala — DISMISSED** [review]
Hi Jiayue-He! Good job changing the ui.R. and server.R files to one app.R file! Just a few general things to consider: 
- I added many comments regarding the style because I was comparing the app side-by-side to the style guide. Make sure you go back in and add any necessary titles and remove and unnecessary details like the colons, strong(), and h4()s for bullet points. 
- Be sure to go through the whole code and add white space around = signs and other symbols where needed. 
- I would ask Dr. Neil and/or Mr. Carey about the huge chunks of code that are commented out and see if you can delete them. 
- Change the order of the tabs and fix the go buttons
- Cite the packages you haven't cited yet. Check if all of them are used with the stale packages function we talked about today
- Make sure your code is explicit everywhere and by this I mean the whole session = session, inputId = "" and so on 
- And for the overall code, I would try to go back when you have free time and fix the alignment and indenting of the code. I noticed that sometimes each line of text underneath the previous line was not consistently indented. For example, some lines are indented twice or thrice while some of the later lines are indented once. I would change them all to one just so we do not go past that 80 character guideline. 

Overall, great job! You did a lot of great work to get the app to where it is now! I hope some of my comments will help!

File: DESCRIPTION [hunk 23]
```diff
@@ -0,0 +1,23 @@
...
+Description: This app will introduce logistic regression model to you and it 
+             allow you to explore Single Logistic Regression with simulated data and the 
+             Empirical Logit Plot with real datasets.
+LearningObjectives: c(
+    "The student will learn to understand the parameters in the Logistic Regression model",
+    "The student will learn to draw the plot of Logistic Regression model with specific dataset."
+  )
+DisplayMode: Normal
+URL: https://psu-eberly.shinyapps.io/Sample_App
+BugReports: https://github.com/EducationShinyAppTeam/Sample_App/issues
```
Comment (lbednarczyk) [inline DESCRIPTION]: The website for URL and BugReports should be updated to say Logistic_Regression instead of Sample_App.

File: app.R [hunk 24]
```diff
@@ -0,0 +1,1215 @@
...
+library(shinyjs) 
+
+#library(discrimARTs)
+library(leaflet)
+library(raster)
+library(DT)
+
+library(RColorBrewer) 
+#library(car)
+#library(rgdal)
```
Comment (lbednarczyk) [inline app.R]: You could delete the lines that have unused packages instead of leaving them commented out.

File: app.R [hunk 25]
```diff
@@ -0,0 +1,1215 @@
...
+              withMathJax(),
+              h4(tags$li("Empirical logit plot is used to check the linearity for datasets.")),
+              div(style="font-size: 1.6em", helpText('$$ {logit ( \\hat p )=log({ \\hat p\\over1-\\hat p})}$$')),
+              h4(tags$li("Deviance Residual and Pearson Residual check the model fit. Best 
+                                                results are no patterns or no extremely large residuals ")),
+              h4(tags$li("Hosmer and Lemeshow test check the goodness of fit in the model 
+                                                where data is divided into recommended 10 groups. The p-value can 
+                                                determine the significance of the result.")),
+              br(),
+              h4(tags$li("Hosmer-Lemeshow Test Statstics")),
```
Comment (lbednarczyk) [inline app.R]: There is a typo in Statistics here.

File: app.R [hunk 26]
```diff
@@ -0,0 +1,1215 @@
...
+              div(style="font-size: 1.6em", helpText('$${\\sum_{i=1}^g}{\\sum_{j=1}^2}{{(obs_{ij} - exp_{ij})^2} 
+                                                                            \\over exp_{ij}}$$')),
+              br(),
+              br(),
+              div(style = "text-align: center",bsButton("start","Go to the overview",
```
Comment (lbednarczyk) [inline app.R]: After you update the order of tabs to be first overview and then prerequisites, the label on this button should be changed to Explore.

File: app.R [hunk 27]
```diff
@@ -0,0 +1,1215 @@
...
+                                                        icon("bolt"),style = "danger",
+                                                        size = "large",class="circle grow"))
```
Comment (lbednarczyk) [inline app.R]: None of the buttons should have the class = circle grow argument.

File: app.R [hunk 28]
```diff
@@ -0,0 +1,1215 @@
...
+                      tableOutput("citable"),
+                      plotOutput("residualPlot", width = "100%",height = "330px")%>% withSpinner(color="#ffa500"),
+                      tags$style(type='text/css', '#lemeshowTest, #obsexp {background-color: rgba(249, 105, 14, 1);  
+                                                      color: yellow; text-align: center}', '#title{color: blackl; padding-left:2.5em; font-size: 22px}'), 
+                      #rgba(219,193,195,0.20)
+                      br(),
+                      h3(strong(id='title', "Hosmer and Lemeshow goodness of fit test")),
+                      #br(),
+                      tableOutput("lemeshowDF"),
+                      tableOutput("obsexpDF"),
```
Comment (lbednarczyk) [inline app.R]: I was getting errors on the page where these outputs were supposed to be shown, the packages ResourceSelection and data.table should be added at the top of the app so that these outputs will run properly.

File: app.R [hunk 29]
```diff
@@ -0,0 +1,1215 @@
...
+                      #verbatimTextOutput("lemeshowTest"),
+                      #verbatimTextOutput("obsexp"),
+                      bsPopover("lemeshowDF"," ","The Hosmer-Lemeshow Test is a goodness of fit test for the logistic model. Here is the result of the Hosmer-Lemeshow Test for ten groups. Number of subgroups, g, usually uses the formula g > P + 1. P is number of covariates. Degree of freedom equals g-2. ", trigger = "hover",placement = "left"),
+                      bsPopover("obsexpDF"," ","There are 10 rows meaning g=10.", trigger = "hover",placement = "left")
+                    )
+                  ),
+                  #set continue button
+                  div(style = "text-align: center",
+                      bsButton(inputId = "go1", label =  "Play the game!", icon("bolt"), style= "danger", size= "large", class='circle grow')
```
Comment (lbednarczyk) [inline app.R]: You could change the label on this button to just say "Play!".

File: app.R [hunk 30]
```diff
@@ -0,0 +1,1215 @@
...
+                      selectInput(inputId="datatable", label="Select Dataset:", 
+                                  choices= c('MedGPA', 'Titanic', 'Leukemia'), 
+                                  selected = 'MedGPA'),
+                      
+                      ####variable options for 'MedGPA' dataset
+                      conditionalPanel(
+                        condition = "input.datatable == 'MedGPA'",
+                        selectInput(inputId="MedYvar", label="Select Response Variable Y",
+                                    choices = c("Acceptance"),
+                                    selected = 'Acceptance'),
```
Comment (lbednarczyk) [inline app.R]: Not sure if this selection is needed since there is only one option for the response variable for each of the possible data sets.

File: app.R [hunk 31]
```diff
@@ -0,0 +1,1215 @@
...
+                      ###number of groups
+                      sliderInput("ngroups", "Number of Groups (Intervals):",
+                                  min = 2, max = 8,
+                                  value = 4, step = 1),
+                      
+                      br()
+                      
+                    ),
+                    mainPanel(
+                      plotOutput("empericalLogitPlot", width = "100%")%>% withSpinner(color="#ffa500")
```
Comment (lbednarczyk) [inline app.R]: This plot wouldn't load until I added the Stat2Data package at the beginning of the code.

**Review by lbednarczyk — DISMISSED** [review]
Good job overall! Other than the comments that I left throughout the code, I would just make sure to go through the code and fix spacing issues and make all of the arguments explicit like Shravani said. Also make sure to add the missing packages so that everything runs smoothly.

**Comment by rpc5102** [comment]
6aaa28de5649b9d7cb9eef64057b7bed1256bd2a was deployed to shinyapps.io to resolve an SSL Expiry error although I suggest further work to clean up.

**Review by Maxinewsu — DISMISSED** [review]
**Recommended Priority: Urgent**
- Issues are all addressed.
- The link of URL and Bugreports on DESCRIPTION is not correct
- Missing screenshot on README.
- Add a citation on the overview page.
- Add a line "This app was last updated by xxx" at the bottom of the overview page.
- Switch the order of Prerequisite and Overview pages.
- The last three referenes on Reference page are not cited using the required format.
- The font size is too big compared to other shiny apps.

**Review by NurulSyafiqah-pixel — CHANGES_REQUESTED** [review]
Based on the current style guide, there are few changes need to be made:

- On the overview page, 
1) the PennState logo should be deleted.
2) the 'About' should be replaced to the formal app title. *e.g. h1("Logistic Regression Analysis")

- On the Prerequisite page:
1) the title 'Logistic Regression Analysis' should be replaced to simply 'Prerequisite'.

- Heading issues:
1) On the app.R, h1() is used more than once. It should be just ONCE that is for the title on the overview page. 
2) Change other headlines that used h1() to their corresponding level (refer to Style Guide chapter 7: 7.1.4 Headers)

- also, the readme file should be updated.

**Review by NurulSyafiqah-pixel — DISMISSED** [review]
All the changes addressed have been settled. Good to go.

**Comment by Maxinewsu** [comment]
@sv101 Just a reminder that please review this app. Many thanks!

File: app.R [hunk 32]
```diff
@@ -0,0 +1,1091 @@
...
+
+# Define UI for App ----
+## Create the app page ----
+ui <- dashboardPage(
+  skin = "yellow",
+  ### Create the app header ----
+  dashboardHeader(
+    title = "Logistic Regression",
+    tags$li(
+      class = "dropdown",
```
Comment (sv101) [inline app.R:57]: The info button doesn't work. It seems to cause the app to close.

**Review by sv101 — CHANGES_REQUESTED** [review]

**Comment by Maxinewsu** [comment]
@sv101 @NurulSyafiqah-pixel Hello, just a reminder that please re-review this app since I have made some changes based on your previous comments. Thanks!

**Review by NurulSyafiqah-pixel — DISMISSED** [review]

**Comment by Maxinewsu** [comment]
@sv101 Could you please help review this app? This was in the app moidified in the summer and I just realized that I need one more approval, thank you!

**Review by Sburke26 — CHANGES_REQUESTED** [review]
Here are a few issues I noticed while reviewing the app:

The description lifecycle badge does not match the badge in the README.
The year in the link of the year badge is one year behind the one in brackets.
The app description is missing the proper heading in the README.
There are small grammatical errors. (Ex. Missing Periods).
The Citation part of the app is missing in the overview page.
Some of the button inputId's could be improved to be more descriptive and representative of their purpose.
Some of the code is messy/missing proper indentation and does not fit the proper style.
The Game tabname was changed to "qqq". I'm not sure if that is the best way to represent the tab.
When a question is incorrect, the first sentence: "Don't give up, try again !" is probably not the best here. After getting a question wrong you can't resubmit or "try again". You can either go to the next question or restart the game. Also, there is a space between the last word and the exclamation point.
There are a couple of messages in the console: "Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0. Please use `linewidth` instead." and "`geom_smooth()` using formula = 'y ~ x'".

**Review by tarynmch — CHANGES_REQUESTED** [review]
Good app, I just noticed a few things. Not all the packages are listed in the references, there is no app description header in the README file, chapter title should be "Categorical Data" rather than "Chapter 10" in the description file, the i icon is more of an overview rather than a instructions, there is no cite as on the overview, and looking at the WAVE report has some instances of small text. Then in the game if you get a question wrong it says "try again" even though you can not try again. I suggest just saying incorrect or implementing a try again option. Within the coding, there are some unnecessary blank lines, on line 996 it has a warning triangle saying that 'disable' should be 'disabled', and I also noticed that the game tab inputID is 'qqq' which is not very descriptive of the purpose.

**Review by tarynmch — CHANGES_REQUESTED** [review]
Good work with updating the app. The UI looks good, but in the server, I noticed that some of the IDs are not camelcase (like gamescore), some things are not individually indented (line 749 and 755), the chapter name should be changed, and there is no alt text for some of the plots and images in the game. The only thing I suggest is to go through the question bank excel and make the capitalization consistent for the choices.
I also noticed that when using the Single Logistic Regression sliders, it said in the console, "glm.fit: fitted probabilities numerically 0 or 1 occurred."

**Comment by Sburke26** [comment]
Thank you for the review.
I made additional fixes to the app.
The "glm.fit: fitted probabilities numerically 0 or 1 occurred." message can't be removed due to the current way the data is simulated. 
I think I added alt text to everything but the logistic plot which uses renderPlotly() instead of renderPlot(). 
Please let me know if you notice anything else.

**Review by tarynmch — CHANGES_REQUESTED** [review]
Looks good. The main thing I noticed was that when you click game it says, "Select your answer from below" but after you click next it says, "pick an answer from below." I would just make sure that is concise. Then make sure randNum is camelcase. I also suggest that for the game, if you want to get to 20 points within 10 questions, you could add a question counter. Otherwise, the app looks great!

**Comment by Sburke26** [comment]
Thank you for the review. I made some changes and added a question counter. Please let me know if you see anything else.

**Review by tarynmch — DISMISSED** [review]
Looks good! The only thing I suggest is that on the Overview page in the instructions, be sure to mention that the New Sample button is within the Explore tab. Otherwise, be sure to get rid of the unnecessary code in the app.R before you merge the branches. Great work.
